docs/src/pages/demos/selects/MultipleSelect.tsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import clsx from 'clsx';
+import { makeStyles, useTheme, Theme } from '@material-ui/core/styles';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import ListItemText from '@material-ui/core/ListItemText';
+import Select from '@material-ui/core/Select';
+import Checkbox from '@material-ui/core/Checkbox';
+import Chip from '@material-ui/core/Chip';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+    maxWidth: 300,
+  },
+  chips: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  chip: {
+    margin: 2,
+  },
+  noLabel: {
+    marginTop: theme.spacing(3),
+  },
+}));
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
+
+const names = [
+  'Oliver Hansen',
+  'Van Henry',
+  'April Tucker',
+  'Ralph Hubbard',
+  'Omar Alexander',
+  'Carlos Abbott',
+  'Miriam Wagner',
+  'Bradley Wilkerson',
+  'Virginia Andrews',
+  'Kelly Snyder',
+];
+
+type ValuesOf<T extends string[]> = T[number];
+type Name = ValuesOf<typeof names>;
+
+function getStyles(name: Name, personName: Name[], theme: Theme) {
+  return {
+    fontWeight:
+      personName.indexOf(name) === -1
+        ? theme.typography.fontWeightRegular
+        : theme.typography.fontWeightMedium,
+  };
+}
+
+function MultipleSelect() {
+  const classes = useStyles();
+  const theme = useTheme();
+  const [personName, setPersonName] = React.useState<Name[]>([]);
+
+  function handleChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    setPersonName((event.target.value as unknown) as Name[]);
```

Comment:
Just curious, was this required because `event.target.value` is typed as `string`, but actually returns `string[]` here?

Reply:
Yeah that's right, I had a search around for handling a multi select event but I couldn't find anything.

Reply:
From my understanding, the `Select` typing might need a change to support raising a `string[]` type for a multi-select, since right now it seems that the typing only raises a `string` for this event.

Reply:
See #15245. It will be changed to `unknown`. The type depends on the children which can't be analyzed by TypeScript (yet?). So we're following the "strict by default" convention.

Reply:
I had a quick play before and changing the type of `SelectInputProps` to something like this seemed to enable more generic typing when the multiple flag is specified.

<details><summary>`SelectInputProps` type change in [SelectInput.d.ts]()</summary>
<p>

```tsx
export type SelectInputProps = {
  autoFocus?: boolean;
  autoWidth: boolean;
  disabled?: boolean;
  IconComponent?: React.ElementType;
  multiple?: boolean;
  inputRef?: (
    ref: HTMLSelectElement | { node: HTMLInputElement; value: SelectInputProps['value'] },
  ) => void;
  MenuProps?: Partial<MenuProps>;
  name?: string;
  onBlur?: React.FocusEventHandler<any>;
  onClose?: (event: React.ChangeEvent<{}>) => void;
  onFocus?: React.FocusEventHandler<any>;
  onOpen?: (event: React.ChangeEvent<{}>) => void;
  open?: boolean;
  readOnly?: boolean;
  renderValue?: (value: SelectInputProps['value']) => React.ReactNode;
  SelectDisplayProps?: React.HTMLAttributes<HTMLDivElement>;
  tabIndex?: number;
  value: string | number | boolean | Array<string | number | boolean>;
  variant?: 'standard' | 'outlined' | 'filled';
} & (
  | {
      native: true;
      onChange?: (event: React.ChangeEvent<HTMLSelectElement>, child: React.ReactNode) => void;
    }
  | {
      native?: false;
      onChange?: (
        event: React.ChangeEvent<React.SelectHTMLAttributes<HTMLSelectElement>>,
        child: React.ReactNode,
      ) => void;
    });
```

</p>
</details>